src/main/java/at/ac/tuwien/damap/conversion/AbstractTemplateExportFunctions.java

```diff
@@ -326,4 +325,143 @@ public void formattingParagraph(List<XWPFParagraph> xwpfParagraphs, String start
             }
         }
     }
+
+    /**
+     * Removes a table inside a document or nested inside another table.
+     * For nested tables, this works only for depth 1.
+     * If the table is not to be found inside the doc or at depth 1, nothing happens.
+     *
+     * @param doc
+     * @param table
+     */
+    public void removeTable(XWPFDocument doc, XWPFTable table) {
+        int pos = doc.getPosOfTable(table);
+        if (pos != -1) {
+            doc.removeBodyElement(pos);
+        } else { // table not found in document -> nestedTable
+            removeNestedTable(doc, table);
+        }
+    }
+
+    /**
+     * Removes a table which is nested inside another table. Max nested depth is 1.
+     *
+     * @param doc
+     * @param table
+     */
+    private void removeNestedTable(XWPFDocument doc, XWPFTable table) {
+        for (XWPFTableCell cell : getAllOuterTableCells(doc)) {
+            for (XWPFTable nestedTable : cell.getTables()) {
+                if (nestedTable.equals(table)) {
+                    int pos = cell.getTables().indexOf(nestedTable);
+                    // dirty hack since POI XWPF does not offer functionality for removing nested tables
+                    try {
+                        Field beField = cell.getClass().getDeclaredField("tables");
+                        beField.setAccessible(true);
+                        ((List<XWPFTable>) beField.get(cell)).remove(pos); // higher level representation
+                    } catch (Exception ignored) {
+                    }
+                    cell.getCTTc().removeTbl(pos); // low level cell representation
+
+                    return;
+                }
+            }
+        }
+    }
+
+    /**
+     * Removes a paragraph inside a document or inside a table.
+     * This will not work for paragraphs inside nested tables.
+     * If the paragraph is not to be found inside the doc or inside non nested tables, nothing happens.
+     *
+     * @param doc
+     * @param paragraph
+     */
+    public void removeParagraph(XWPFDocument doc, XWPFParagraph paragraph) {
```

Comment:
Is this method used anywhere?

Reply:
No I wrote it in an earlier attempt on solving this and kept it in since it could be useful in the future.

Reply:
If it is not used, it should be removed

Reply:
Done